### DISCUSSION TITLE
Opening foxglove studio and setting a certain port as default data source through terminal.

### DISCUSSION THREAD

**Discussion by LaiWeiQuan** [foxglove/community#813] [General]
Hello, I'm curious if there's a method to launch Foxglove and specify a default port for the data source from the terminal. This would allow visualization process to be automated through command lines.

**Comment by jtbandes** [comment]
You should be able to pass a [`foxglove://` link](https://foxglove.dev/docs/studio/building-and-sharing-links) as a command line argument.

**Reply by LaiWeiQuan** [reply]
I see. So if I have a data source ws://localhost:9090
Do I just do  `foxglove://open?ds=ws://localhost:9090`

**Reply by achim-k** [reply]
Almost, it would be `foxglove://open?ds=rosbridge-websocket&ds.url=ws://localhost:9090` (rosbridge) or `foxglove://open?ds=foxglove-websocket&ds.url=ws://localhost:9090` (foxglove websocket)
